server/backends/blobstore/gcs/gcs.go

```diff
@@ -220,7 +221,13 @@ func (g *GCSBlobStore) ConditionalWriter(ctx context.Context, blobName string, o
 
 	cwc := ioutil.NewCustomCommitWriteCloser(ow)
 	cwc.CommitFn = func(int64) error {
-		return ow.Close()
+		err := ow.Close()
+		if gerr, ok := err.(*googleapi.Error); ok {
+			if gerr.Code == http.StatusPreconditionFailed {
```

Comment:
ick, that's as detailed as it gets, huh?

Reply:
this is the same code as before just moved over, but yeah kind of meh.